test/tensor_reorder.cpp

```diff
@@ -488,11 +518,16 @@ struct reorder_test : reorder_base
                                          nullptr,
                                          nullptr);
             EXPECT(status == CL_SUCCESS);
+            clReleaseMemObject(src_dev);
+            }
 #elif MIOPEN_BACKEND_HIP
+            if(reorder_sol->IsSkippable()){
             EXPECT(hipMemcpy(t_dst_gpu.data.data(),
-                             dst_dev,
+                             src_dev,
```

Comment:
why copy `src_dev` to `t_dst_gpu`?

Reply:
when `IsSkippable() == true`, the reorder driver won't call reorder operation on GPU, and compare the origin tensor with CPU reordered tensor.

Reply:
in `batched_transpose`, will still call the kernel even when `IsSkippable() == true`, if you actually call this kernel. The `IsSkippable()` is for the caller to optimize this process when `IsSkippable() == true`. In Other words, the gpu kernel should be OK to function well even when `IsSkippable() == true`. Is the design of generic transpose that gpu kernel will not compute correctly in such case?

Another thing is https://github.com/ROCmSoftwarePlatform/MIOpen/blob/c223c9461f198dddbd198b32f5a6b98a3c1dc27c/test/tensor_reorder.cpp#L521
`src_dev` will only be released here, when `IsSkippable() == true`, lead to memory leak.

Reply:
I see. The ctest logic has been updated in new commits.